Framework/Source/Sample.cpp

```diff
@@ -40,504 +40,695 @@
+		startCheck = startCheck + ("Total Used Virtual Memory By All Processes : " + startTUVM_B + " bytes, " + startTUVM_MB + " MB. \n");
+		startCheck = startCheck + ("Virtual Memory used by this Process : " + startCUVM_B + " bytes, " + startCUVM_MB + " MB. \n \n");
+
+		//	Get the Strings for the Memory in Bytes - End Frame
+		std::string endTVM_B = std::to_string(memoryCheckRange.endCheck.pTotalVirtualMemory);
+		std::string endTUVM_B = std::to_string(memoryCheckRange.endCheck.pTotalUsedVirtualMemory);
+		std::string endCUVM_B = std::to_string(memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory);
+
+		std::string endTVM_MB = std::to_string(memoryCheckRange.endCheck.pTotalVirtualMemory / (1024 * 1024));
+		std::string endTUVM_MB = std::to_string(memoryCheckRange.endCheck.pTotalUsedVirtualMemory / (1024 * 1024));
+		std::string endCUVM_MB = std::to_string(memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory / (1024 * 1024));
+
+		std::string endCheck = "At the End Frame, " + std::to_string(memoryCheckRange.endCheck.pFrame) + ", : \n ";
+		endCheck = endCheck + ("Total Virtual Memory : " + endTVM_B + " bytes, " + endTVM_MB + " MB. \n");
+		endCheck = endCheck + ("Total Used Virtual Memory By All Processes : " + endTUVM_B + " bytes, " + endTUVM_MB + " MB. \n");
+		endCheck = endCheck + ("Virtual Memory used by this Process : " + endCUVM_B + " bytes, " + endCUVM_MB + " MB. \n \n");
+
+		//	Compute the Difference Between the Two.
+		std::string differenceCheck = "Difference : \n";
+		long long difference = 0;
+
+		{
+			difference = (long long)memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory - (long long)memoryCheckRange.startCheck.pCurrentlyUsedVirtualMemory;
+			differenceCheck = differenceCheck + std::to_string(difference) + "\n \n";
+		}
+
+		//	Key string for difference.
+		std::string keystring = std::to_string(memoryCheckRange.startCheck.pFrame) + " " + std::to_string(memoryCheckRange.endCheck.pFrame) + " " + (startCUVM_B)+" " + (endCUVM_B)+" " + std::to_string(difference) + " \n";
+
+		//	Get the name of the current program.
+		std::string filename = getExecutableName();
+
+		// Now we have a folder and a filename, look for an available filename (we don't overwrite existing files)
+		std::string prefix = std::string(filename) + ".MemoryFrameCheck";
+		std::string executableDir = getExecutableDirectory();
+		std::string txtFile;
+
+
+		//	Get an available filename.
+		if (findAvailableFilename(prefix, executableDir, "txt", txtFile))
+		{
+			//	Output the memory check.
+			std::ofstream of;
+			of.open(txtFile);
+			of << keystring;
+			of << differenceCheck;
+			of << startCheck;
+			of << endCheck;
+			of.close();
+		}
+		else
+		{
+			//	Log Error.
+			logError("Could not find available filename when checking memory.");
+		}
+
+	}
+
+	//	Write the Memory Check Range, in terms of Time to a file. Outputs Difference, Start and End Times and Memories.
+	void Sample::writeMemoryTimeRange(const MemoryCheckRange & memoryCheckRange)
```

Comment:
This function looks very similar to the one above. Can you abstract them for better code sharing?